src/mongocxx/options/find.cpp

```diff
@@ -211,6 +214,179 @@ const stdx::optional<class read_preference>& find::read_preference() const {
     return _read_preference;
 }
 
+find& find::convert_all_modifiers() {
+    if (!_modifiers) {
+        return *this;
+    }
+
+    for (auto&& ele : _modifiers->view()) {
+        if (ele.key() == stdx::string_view("$comment")) {
+            if (ele.type() != bsoncxx::type::k_utf8) {
+                throw exception{error_code::k_invalid_parameter,
```

Comment:
I've posted a fixup commit that uses `mongocxx::logic_error` here instead of `mongocxx::exception`.